platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -76,6 +111,39 @@ namespace android {
         source.as<mbgl::style::GeoJSONSource>()->GeoJSONSource::setGeoJSON(GeoJSON(features));
     }
 
+    void GeoJSONSource::setFeatureCollectionAsync(jni::JNIEnv& env, jni::Object<geojson::FeatureCollection> jFeatures,
+                                                  jni::Object<OnGeoJsonSourceLoadedListener> jListener) {
+        // If another update is running, log an error
+        if (callback) {
+            mbgl::Log::Error(mbgl::Event::JNI, "Error setting GeoJSON: another asynchronous update of this source is being processed.");
+            return;
+        }
+
+        callback = std::make_unique<Actor<FeatureConverter::Callback>>(
+                *Scheduler::GetCurrent(),
+                [this](GeoJSON geoJSON) {
+                    android::UniqueEnv _env = android::AttachEnv();
+
+                    // Update the core source
+                    source.as<mbgl::style::GeoJSONSource>()->GeoJSONSource::setGeoJSON(geoJSON);
+
+                    collectionRef.release();
```

Comment:
This doesn't actually clean up any resource: `release()` tells the `unique_ptr` that you're manually taking ownership of the resource. You want `reset()` instead.

https://en.cppreference.com/w/cpp/memory/unique_ptr/release